File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int count_letters(char letter);
int count_words(char letter);
int count_sentences(char letter);

int main(void)
{
    string text = get_string("Text:");
    //printf("text[0]: %c\n", text[0]);
    int len = strlen(text);
    //printf ( "string length  = %d\n", len );

    char letter;
    int count=0;
    float total_letter=0;
    float t_letter=0;
    float t_word=0;
    float total_words=0;
    float t_sentence=0;
    float total_sentences=0;
    for(int i=0; i<=len; i++)
    {
        letter = text[i];
        //printf("text[%d]: %c", i, letter);
        t_letter = count_letters(letter);
        //printf("t_letter = %i", t_letter);
        total_letter = total_letter+t_letter;

        t_word = count_words(letter);
        total_words=total_words+t_word;

       if ((letter == 33) || (letter == 46) || (letter == 63))
       {
           if((text[i-1] > 64 && text[i-1] < 91) || (text[i-1] > 64 && text[i-1] < 91) || (text[i-1] > 64 && text[i-1] < 91))
           {
               break;
           }
           else
           {
                t_sentence = count_sentences(letter);
                total_sentences = total_sentences + t_sentence;
           }
       }
    }

    total_words+=1;
    // printf("\n");
    // printf("total_letter: %f\n", total_letter);
    // printf("total_words: %f\n", total_words);
    // printf("total_sentences: %f\n", total_sentences);


   float pre_L = total_letter/total_words;
   //printf("pre_L = %f\n", pre_L);
   float L = pre_L*100;

   float pre_S = (total_sentences/total_words);
   //printf("pre_S = %f\n", pre_S);
   float S = pre_S*100;

   //printf("L = %f\n", L);
  // printf("S = %f\n", S);

   float Index = round((0.0588 * L) - (0.296 * S) - 15.8);
    int grade = Index;

    if(Index<1)
    {
        printf("Before Grade 1\n");
    }

    else if(Index>=16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade);

    }
}

int count_letters(char letter)
{
    int count = 0;
    if ((letter > 64 && letter < 91 ) || (letter > 96 && letter < 123 ))
    {
        count = count + 1;
        //printf(" count: %i\n", count);
    }
    return count;
}

int count_words(char letter)
{
    int count = 0;
    if (letter == 32)
    {
        count = count + 1;
        //printf(" count: %i\n", count);
    }
    return count;
}

int count_sentences(char letter)
{
    int count = 0;
    if ((letter == 33) || (letter == 46) || (letter == 63))
    {
        count = count + 1;
        //printf(" count: %i\n", count);
    }
    return count;
}

File: readability.py
from cs50 import get_string

    
def main():
    text = get_string("Text: ")
    #print(len(text))
    
    t_letter=0
    total_letter=0
    t_word=0
    total_words=1
    t_sentence=0
    total_sentences=0

   

    for i in range(0, len(text)):
        letter = text[i];
        #print(letter)
        t_letter=count_letters(letter);
        #print(f"t_letter:{t_letter}")
        total_letter = total_letter+t_letter
         #print(f"total_letter: {total_letter}")
       
        t_word = count_words(letter);
        total_words = total_words+t_word
        
        if((letter == '.') or (letter == '?') or (letter == '!')):
            pre_letter1 = text[i-1]
            pre_letter2 = text[i-2]
            if(pre_letter1.isupper() == True or pre_letter2.isupper() == True):
                continue
            else:
                t_sentence = count_sentences(letter);
                total_sentences = total_sentences + t_sentence
    
    
    #print(f"total_letter: {total_letter}")
    #print(f"total_words: {total_words}")
    #print(f"total_sentences: {total_sentences}")
    
    pre_L = float(total_letter/total_words);
    L = float(pre_L*100);
    
    pre_S = float(total_sentences/total_words);
    S = float(pre_S*100);
    
    Index = round((0.0588 * L) - (0.296 * S) - 15.8);
    grade = int(Index);

    if(Index<1):
        print("Before Grade 1")

    elif(Index>=16):
        print("Grade 16+")
        
    else:
        print(f"Grade {grade}")


def count_letters(letter):
    count = 0
    #print(type(letter))
    if ((letter >= 'a' and letter <= 'z') or (letter >= 'A' and letter <= 'Z')):
        count = count + 1
        #print(f"count of count_letters: {count}")
    return count
    
def count_words(letter):
    #print(f"in fun2")
    count2 = 0
    if letter.isspace()==True:
        count2 = count2 + 1
        #print(f"COUNT of count_words: {count2}")
    return count2

def count_sentences(letter):
    count3 = 0
    if((letter == '.') or (letter == '?') or (letter == '!')):
        count3 = count3 + 1
       # print(f"Count of count_sentences : {count3}")
    return count3

if __name__ == "__main__":
    main()